streams/sidecar_file.proto

```diff
@@ -53,15 +55,43 @@ message TransactionSidecarRecord {
    */
   Timestamp consensus_timestamp = 1;
 
+  SidecarGenerationType generation_type = 2;
+
   /*
   * List of sidecar types.
   * In future there will be other categories.
   */
   oneof sidecar_records {
-    ContractStateChanges state_changes = 2;
+    ContractStateChanges state_changes = 3;
+    ContractActions actions = 4;
+    ContractBytecode bytecode = 5;
   }
 }
 
+/**
+* The generation type described by the TransactionSidecarRecord.
+*/
+enum SidecarGenerationType {
+  /**
+   * default non-value.
+   */
+  SIDECAR_GENERATION_TYPE_UNKNOWN = 0;
+
+  /**
+  * Sidecar generated from migration.
+  */
+  MIGRATION = 1;
+
+  /**
+  * Sidecar generated from trace.
+  */
+  TRACE = 2;
```

Comment:
I imagine in the future we might put other, non-traceability information in the sidecar like errata records so we should name this something more general. Or we can just change it to a `bool migration` which might be clearer.

Whichever data type, we should choose the default value to be the non-migration value so it doesn't incur the cost of writing extra bytes indefinitely to the file for the one-time migration. If we use `bool migration` then the default value is false and would not serialize the field for the majority of record files.

Reply:
DONE